airbyte-integrations/connectors/source-snowflake/unit_tests/integration/test_discover.py

```diff
@@ -0,0 +1,226 @@
+        find_template(file_name, __file__),
+        datafield,
+        record_id_path=id_path,
+        record_cursor_path=cursor_path
+    )
+
+
+def _given_table_catalog(http_mocker: HttpMocker) -> None:
+    http_mocker.post(
+        table_request().with_show_catalog().with_requestID(_REQUESTID).build(),
+        snowflake_response("check_connection").with_record(a_snowflake_response("check_connection")).build()
+    )
+
+
+def _given_table_with_primary_keys(http_mocker: HttpMocker, table) -> None:
+    http_mocker.post(
+        table_request().with_table(table).with_show_primary_keys().with_requestID(_REQUESTID).build(),
+        snowflake_response("primary_keys").with_record(a_snowflake_response("primary_keys")).build()
+    )
+
+def _given_read_schema(http_mocker: HttpMocker, table) -> None:
+    http_mocker.post(
+        table_request().with_table(table).with_get_schema().with_requestID(_REQUESTID).build(),
+        snowflake_response("response_get_table", JsonPath("$")).with_record(
+            a_snowflake_response("response_get_table", JsonPath("$"))).build()
+    )
+
+def _given_get_timezone(http_mocker: HttpMocker) -> None:
+    http_mocker.post(
+        table_request().with_requestID(_REQUESTID).with_timezone().build(),
+        snowflake_response("timezone", JsonPath("$.'data'")).build())
+
+
+def _discover(
+        config_builder: ConfigBuilder,
+        sync_mode: SyncMode,
+        state: Optional[List[AirbyteStateMessage]] = None,
+        expecting_exception: bool = False
+) -> EntrypointOutput:
+        catalog = _catalog(sync_mode)
+        config = config_builder.build()
+        return discover(_source(catalog, config, state), config, expecting_exception)
+
+@mock.patch("source_snowflake.streams.snowflake_parent_stream.uuid.uuid4", return_value=_REQUESTID)
+@mock.patch("source_snowflake.source.SnowflakeJwtAuthenticator")
+class DiscoverTest(TestCase):
+
+    @classmethod
+    def _a_table_record(cls):
+        return a_snowflake_response("check_connection", JsonPath("$.'data'.[*]"), id_path=JsonPath("$.[1]"))
+
+    @HttpMocker()
+    def test_discover_no_pagination(self,  uuid_mock, mock_auth, http_mocker: HttpMocker)->None:
+            
+        
+        _given_get_timezone(http_mocker)
+       
+        for i in ["TEST_TABLE_2","TEST_TABLE_1"]:
+            _given_table_with_primary_keys(http_mocker, i)
+            _given_read_schema(http_mocker,i)
+       
+
+        http_mocker.post(
+        table_request().with_show_catalog().with_requestID(_REQUESTID).build(),
+        snowflake_response("check_connection", JsonPath("$.'data'"))
+        .with_record(
+            self._a_table_record().with_id("TEST_TABLE_1")
+        )
+        .with_record(
+            self._a_table_record().with_id("TEST_TABLE_2"))
+        .build()
+        )
+        
+        
+        output= _discover(_config(), sync_mode=SyncMode.full_refresh)
+        assert len(output.catalog.catalog.streams)
```

Comment:
Is it too precise at this level to `assert len(output.catalog.catalog.streams) == 2`?

Reply:
i think that's what i wanted to do. i made a mistake